main.py

```diff
@@ -113,17 +115,17 @@ def get_metrics(task, threshold):
     best_metric = []
     if task == "regression":
         metrics_reg = [MSE(), CorrCoef()]
-        best_metric = metrics_reg[-1]
     elif task == "classification":
         metrics_cla = [BCE(), Recall(threshold),
                        Specificity(threshold), AUROC()]
-        best_metric = metrics_cla[-1]
     elif task == 'both':
         metrics_reg = [MSE(), CorrCoef()]
         metrics_cla = [BCE(), Recall(threshold),
                        Specificity(threshold), AUROC()]
-        best_metric = metrics_cla[-1]
-
+    try:
+        best_metric = getattr(metrics, best_metric_choice)()
```

Comment:
The old code was using the reference to an existing metric object as the best metric, whereas this new code is instantiating a new metric object. I think this may cause some issues downstream because the best metric variable is likely not updated in the same way the other metrics are. I think the safest thing here would be to look for an instance of the best metric class in the metrics arrays and use that reference

Reply:
makes sense. made the changes. good catch